### ISSUE TITLE
How to produce experiments for BalanceSoftmax+Decouple Training?

### ISSUE THREAD

**Issue by hongxin001** [jiawei-ren/BalancedMetaSoftmax-Classification#11]
I found there is a config for BalanceSoftmax+Decouple Training in CIFAR-10-LT and it trains a classifier with BS based on the feature extractor from standard training. However, I cannot replicate the result reported in the paper using the config. Is there any different?

**Comment by jiawei-ren** [comment]
Please provide more details, e.g., which config did you use, what results did you get, and the training logs.

**Comment by hongxin001** [comment]
For example, on Places-LT, I use the config: ./config/Places_LT/decouple_balanced_softmax.yaml and I got 35.1. The training log is as below: 
[places_dt_bs.log](https://github.com/jiawei-ren/BalancedMetaSoftmax-Classification/files/7533592/places_dt_bs.log)

**Comment by jiawei-ren** [comment]
I notice that the batch size in your log is 128 while the batch size in the config is 256.

**Comment by hongxin001** [comment]
Thanks for your advice.

**Comment by jiawei-ren** [comment]
We made a typo in Places-LT's config. The sampler option should be 'null' instead of 'ClassAwareSampler'. Thank you for raising the issue. We will fix the config asap. 

Are you still unable to reproduce CIFAR-10-LT's result?

**Comment by hongxin001** [comment]
I will have a try

**Comment by hongxin001** [comment]
I found that the sampler option in CIFAR-10-LT is also ClassAwareSampler. Should I change it to 'null'?

**Comment by jiawei-ren** [comment]
I don't see any ClassAwareSampler in CIFAR-10-LT's configs. Would you mind specifying the config name?

**Comment by hongxin001** [comment]
I see. That's my fault.

**Comment by hongxin001** [comment]
For CIFAR-10-LT, I got 78.9.

**Comment by hongxin001** [comment]
The result on Places-LT is still not good. Could you have a try?

**Comment by jiawei-ren** [comment]
May I have a look at the new training log?

**Comment by hongxin001** [comment]
[places_dt_bs (1).log](https://github.com/jiawei-ren/BalancedMetaSoftmax-Classification/files/7533985/places_dt_bs.1.log)

**Comment by jiawei-ren** [comment]
We only ran and reported the 'BALMS' setting on Places-LT in the paper. In fact, the config is not part of the paper, therefore I am not sure what is the result that the config supposes to give. Although the low-shot accuracy in your training log seems very low (even lower than the baseline), the result you get could be the right one. 

I will remove the config to avoid further confusion. If you are still interested in using Balanced Softmax in Places-LT, I would recommend end-to-end training instead of decoupled training. Recent papers [1, 2] report very competitive results (39.4 top-1 acc) using 'end-to-end + Balanced Softmax' on Places-LT.

[1] Disentangling Label Distribution for Long-tailed Visual Recognition
[2] Test-Agnostic Long-Tailed Recognition by Test-Time Aggregating Diverse Experts with Self-Supervision

**Comment by hongxin001** [comment]
I see. Thanks for your explanation and the introduced papers. Although not working in this setting, Balanced Softmax is still a valuable method in this area.
